test/api/projects_api_test.rb

```diff
@@ -11,10 +11,23 @@ def app
   end
 
   def test_can_get_projects
-    user = FactoryBot.create(:user, :student, enrol_in: 0)
-    get with_auth_token('/api/projects', user)
+    # Create a new dummy unit
+    unit = FactoryBot.create(:unit)
+
+    # Get a new project from unit
+    new_project = unit.active_projects.first
+    student = new_project.student
+
+    # Perform Get   new_project.main_convenor_user
+    get with_auth_token "/api/projects/#{new_project.id}", new_project.student
+    actual_project = last_response_body
+    
+    # Check if the call success
     assert_equal 200, last_response.status
-  end
+
+    # Check the returned details match as new_project
+    assert_equal actual_project['project_id'], new_project.id
```

Comment:
You should use `assert_json_matches_model` and compare other things than just `project_id`. You can try` puts(last_response_body)` and `puts(new_project)` to check which attributes to compare

Reply:
Please check for other things than just `project_id`. This one has not been resolved yet...